enclave/core/sgx/calls.c

```diff
@@ -1105,6 +1125,12 @@ void __oe_handle_main(
     uint64_t* output_arg1,
     uint64_t* output_arg2)
 {
+    /* Get pointer to the thread data structure */
+    oe_sgx_td_t* td = td_from_tcs(tcs);
+
+    /* Stitch the stack. Pass the caller's frame for fix up. */
```

Comment:
Worth mentioning it the comment that oe_enter assembly code initially stitches the stack to the actual host frame when ENCLU(EENTER) is invoked. This call updates the stitch to point to a host-specified frame instead.